src/Strings.h

```diff
@@ -0,0 +1,388 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Details {
+
+template<typename IteratorT>
+inline std::vector<IteratorT> IteratorVectorGenerator(IteratorT const & begin,
+                                                     IteratorT const & end) {
+    std::vector<IteratorT> wordIterVector;
+    wordIterVector.reserve(static_cast<unsigned int>(std::distance(begin, end) + 1));
+
+    for (IteratorT strIter = begin; strIter != end; ++strIter)
+        wordIterVector.emplace_back(std::move(strIter));
+    wordIterVector.emplace_back(std::move(end));
+
+    return wordIterVector;
+}
+
+template <
+    typename IteratorT,
+    typename IsDelimiterT
+>
+inline void Parse(IteratorT const &begin,
+                  IteratorT const &end,
+                  IsDelimiterT const &isDelimiter,
+                  std::function<void (IteratorT&, IteratorT&)> const &callback) {
```

Comment:
>IteratorT&, IteratorT& [](start = 38, length = 22)

Need an explanation or a doc for these callbacks

Reply:
the callbacks are functions to do the string decoration(casing and trimming), as from a previous talk with David that we apply split(parse, ngram...) function before decorate functions